ansible/roles/azure/files/cloud-init.yml

```diff
@@ -0,0 +1,23 @@
+#cloud-config
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# NOTE: do not modify the first line in this file - it is a mandatory
+# header to designate the contents as a cloud-init configuration
+#
+# Upgrade the instance on first boot:
+package_upgrade: true
```

Comment:
We've previously discussed upgrading instances in #408 and again in #411. It feels a bit mischievous to sneak this in under the header of "Use cloud-init" when the previous issues that explicitly discussed updating packages was rejected in favor of updating to use a newer instance instead. I think this should default to `false`, as discussed in those previous issues.

Reply:
@ctubbsii this is to unblock deployments on Azure. As it stands right now, launching a cluster on Azure, with either CentOS 7.5 or 7.9 (and presumably any versions in between) fails because the ca-certificates package on the OS is outdated and consequently the Ansible `get_url` fails.

In the previous PRs, the legitimate concern raised by you was that the upgrade packages step would apply to all cluster types. At the time it seemed that bumping up the image to 7.9 would solve the problem, but unfortunately, that issue was not really resolved.

This PR is conditional to Azure cluster launches and is a must-have, as there are no other more recent updated CentOS 7.x images on Azure. Building a custom patched image and using it for Muchos is time-consuming and not everyone may know how to do that, either. There is no 'mischievous' intent here - this fix is scoped to address the specific problem for Azure based deployments.

Reply:
> This PR is conditional to Azure cluster launches and is a must-have,

I don't think the basis of my previous thoughts were specifically about whether it would apply outside the limited scope of the Azure cluster, but that the idea itself implies that a user consciously configures an out-of-date image, and then has to deal with the wait time to bring it up-to-date, when a much more elegant solution is to just use an up-to-date image from the start.

You say it's a "must-have", but for whom? The problem can be solved in several ways, not the least of which is to change the default to use something newer than an OS initially composed way back in 2013.

> as there are no other more recent updated CentOS 7.x images on Azure.

That seems easily fixable... or avoidable. Why does the default have to be CentOS 7? Why not CentOS 8, CentOS Stream 8 or 9, or Fedora 35, or RHEL 8? The problem of no more recent updated "official" CentOS 7 images seems like a problem only if we make it a problem.

> Building a custom patched image and using it for Muchos is time-consuming and not everyone may know how to do that, either.

Nobody expects everyone to do that. I'm not sure how it works on Azure, but on EC2, community images are shared for everybody to reuse, so it only needs to be done once for many people to benefit from it. I'd be surprised if there weren't already multiple updated CentOS 7 community images available on Azure, unless Azure just doesn't support that.

But, even that's not necessary. A different OS, a newer OS, is also a possible solution. If we keep patching to force an 8 year old CentOS 7 to keep working, rather than change the default image to something newer, so that we default to testing on newer operating systems likely to be used in future Accumulo and Fluo deployments, then I think we're solving the problem wrong.

> There is no 'mischievous' intent here - this fix is scoped to address the specific problem for Azure based deployments.

Perhaps that word isn't the right one. I don't believe this is nefarious, deceptive, or anything like that. The description was clear about the intent behind the proposed change and I did not mean to imply otherwise. "Mischievous" seemed like a more "playful" term to use, since the PR seemed like it was tip-toeing around the previous discussions rather than directly addressing them. I did not mean any offense by my use of that word, but I am sorry if I caused any.

Reply:
Thanks for clarifying. The bump to use a more recent OS is certainly the right way forward but needs some work to test with to select a default. I'm sure you will agree that making CentOS 8.x the default is not a good idea, given it is [EOL](https://www.centos.org/centos-linux-eol/) by the end of this month. The other alternatives are definitely worth looking at.

For the interim, I would like to proceed with this change, to unblock users like me who are using Azure. I will look at the alternative, modern OS default, early next year.